content/blog/parsnip-0-1-2/index.Rmd

```diff
@@ -0,0 +1,279 @@
+
+* Tree-based models, such as CART, C5.0, or random forests, don't require binary indicator variables since their splitting methods can create groups of categories. For a function like `ranger::ranger()`, the formula does _not_ create indicators; the factor predictors are left as factors. 
+
+* Naive Bayes models would rather have the predictors in their native format so that their conditional distributions are estimated using discrete probability distributions. 
+
+* Multi-level models, such as mixed models or Bayesian hierarchical models, would prefer that the columns that are associated with random effects (e.g. subject) to remain factors. 
+
+When `parsnip::fit()` is used with a modeling function that takes a formula, the formula is directly passed to the underlying model function (without processing the data). The resulting model is the same as what the underlying model would have produced. For example: 
+
+```r
+# Using older versions of: 
+library(parsnip)
+library(ranger)
+library(modeldata)
+
+data(penguins)
+penguins <- na.omit(penguins)
+
+rf_spec <-
+  rand_forest() %>%
+  set_engine("ranger", seed = 1221) %>%
+  set_mode("regression")
+
+rf_spec %>%
+  fit(body_mass_g ~ species + island, data = penguins)
+```
+
+```
+## parsnip model object
+## 
+## Fit time:  24ms 
+## Ranger result
+## 
+## Call:
+##  ranger::ranger(formula = formula, data = data, seed = ~1221, num.threads = 1, verbose = FALSE) 
+## 
+## Type:                             Regression 
+## Number of trees:                  500 
+## Sample size:                      333 
+## Number of independent variables:  2 
+## Mtry:                             1 
+## Target node size:                 5 
+## Variable importance mode:         none 
+## Splitrule:                        variance 
+## OOB prediction error (MSE):       224771.2 
+## R squared (OOB):                  0.6533301
+```
+
+```r
+ranger(body_mass_g ~ species + island, data = penguins, seed = 1221)
+```
+
+```
+## Ranger result
+## 
+## Call:
+##  ranger(body_mass_g ~ species + island, data = penguins, seed = 1221) 
+## 
+## Type:                             Regression 
+## Number of trees:                  500 
+## Sample size:                      333 
+## Number of independent variables:  2 
+## Mtry:                             1 
+## Target node size:                 5 
+## Variable importance mode:         none 
+## Splitrule:                        variance 
+## OOB prediction error (MSE):       224771.2 
+## R squared (OOB):                  0.6533301
+```
+
+(Note the `Number of independent variables:  2`). 
+
+However, the workflows package does process the data before giving it to the modeling function. In this case, indicators were produced when a formula is used. As a result, instead of two independent variables, the `species` was expanded and five predictor columns are given to the model: 
```

Comment:
```suggestion
However, the workflows package _does_ process the data before giving it to the modeling function. In this case using the previous version of these packages, indicators were produced when a formula was used. As a result, instead of two predictor columns, the `species` variable was expanded and five predictor columns are given to the model: 
```